### PR TITLE
chore: refactor gradle scripts

### PR DESCRIPTION
**Pull request by RyuNen344**
## Issue
- no issue

## Overview (Required)
- Ktorfit plugin breaks arch filter, so I fixed it
- create version catalog dsl, to remove redundant `get()` call
  - ktorfit ksp plugin didn't match ktorfit library, so replaced plugin to match version catalog

## Links
- 

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/KmpIosPlugin.kt [hunk 1]
```diff
@@ -33,25 +24,18 @@ class KmpIosPlugin : Plugin<Project> {
...
-                        "iosX64",
-                        "iosSimulatorArm64",
-                    )
                     create("iosMain") {
                         dependsOn(getByName("commonMain"))
-                    }
-                    iosSourceSets.forEach { iosSourceSet ->
-                        maybeCreate(iosSourceSet + "Main")
-                            .dependsOn(getByName("iosMain"))
+                        maybeCreate("iosArm64Main").dependsOn(this)
```
Comment (RyuNen344) [inline build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/KmpIosPlugin.kt:29]: like official doc

https://kotlinlang.org/docs/multiplatform-hierarchy.html#target-shortcuts

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/AndroidComposePlugin.kt [hunk 2]
```diff
@@ -10,25 +10,22 @@ class AndroidComposePlugin : Plugin<Project> {
...
-                implementation(libs.findLibrary("androidxLifecycleLifecycleRuntimeKtx"))
-                implementation(libs.findLibrary("androidxActivityActivityCompose"))
-                testImplementation(libs.findLibrary("junit"))
-                testImplementation(libs.findLibrary("androidxTestExtJunit"))
-                testImplementation(libs.findLibrary("androidxTestEspressoEspressoCore"))
-                testImplementation(libs.findLibrary("composeUiTestJunit4"))
-                debugImplementation(libs.findLibrary("composeUiTooling"))
-                debugImplementation(libs.findLibrary("composeUiTestManifest"))
-                lintChecks(libs.findLibrary("composeLintCheck"))
+                implementation(libs.library("androidxCoreKtx"))
```
Comment (takahirom) [inline build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/AndroidComposePlugin.kt:16]: Looks great!

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.gradle.plugin.use.PluginDependency
+
+internal val Project.libs: VersionCatalog
+    get() = extensions.getByType<VersionCatalogsExtension>().named("libs")
+
+internal fun VersionCatalog.version(name: String): String {
+    return findVersion(name).get().requiredVersion
+}
+
+internal fun VersionCatalog.library(name: String): MinimalExternalModuleDependency {
```
Comment (takahirom) [inline build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt:18]: Now I see why we can do this 👀

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt
Comment (takahirom) [inline build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt:1]: We are using `〜Dsl` for extensions how about using VersionCatalogDsl?

**Review by takahirom — APPROVED** [review]
Looks great!

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt (hunk 3 above)
Comment (RyuNen344) [reply build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt:18]: we can call version catalog more shortly and not depends on Java/Optional :yay:

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt
Comment (RyuNen344) [reply build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/VersionCatalog.kt:1]: OK!! I gonna rename file.

**Pull request merged**